jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/LogStreamActor.kt

```diff
@@ -70,21 +70,35 @@ sealed class LogStreamActor(
                     val items = loadMore(nextBackwardToken, saveBackwardToken = true)
                     withContext(edtContext) { table.listTableModel.items = items + table.listTableModel.items }
                 }
-                is Message.LOAD_INITIAL -> loadInitial()
-                is Message.LOAD_INITIAL_RANGE -> loadInitialRange(message.startTime, message.duration)
-                is Message.LOAD_INITIAL_FILTER -> loadInitialFilter(message.queryString)
+                is Message.LOAD_INITIAL -> {
+                    loadInitial()
+                    // make sure the scroll pane is at the top after loading. Needed for Refresh!
+                    val rect = table.getCellRect(0, 0, true)
+                    table.scrollRectToVisible(rect)
+                }
+                is Message.LOAD_INITIAL_RANGE -> {
+                    loadInitialRange(message.previousEvent.timestamp, message.duration)
+                    val item = table.listTableModel.items.firstOrNull { it == message.previousEvent }
+                    val index = table.listTableModel.indexOf(item).takeIf { it > 0 } ?: return
+                    table.setRowSelectionInterval(index, index)
+                    val rect = table.getCellRect(index, 0, true)
+                    table.scrollRectToVisible(rect)
+                }
+                is Message.LOAD_INITIAL_FILTER -> {
+                    loadInitialFilter(message.queryString)
+                }
             }
         }
     }
 
     protected suspend fun loadAndPopulate(loadBlock: suspend () -> List<LogStreamEntry>) {
         try {
+            tableLoading()
             val items = loadBlock()
-            withContext(edtContext) {
-                table.listTableModel.items = items
-            }
+            table.listTableModel.items = items
             table.emptyText.text = emptyText
         } catch (e: ResourceNotFoundException) {
+            table.isVisible = false
```

Comment:
Do we need to make visible at like 96? Not 100% sure how this interacts with the refresh action